### PR TITLE
executor: Update history stats with write path api

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by zimulala — APPROVED** [review]
LGTM

File: executor/adapter.go [hunk 1]
```diff
@@ -1504,84 +1505,97 @@ func resetCTEStorageMap(se sessionctx.Context) error {
...
 	}
-	return false
+	return true
 }
 
-func recordHistoryStats(sql string, tableID int64, planID int, nullHashCode bool, hashCode uint64, estRows float64, actualRows int64) {
+func recordHistoryStats(statsHandle *handle.Handle, startTS uint64, tableID int64, hashCode uint64, actualRows int64, totalTableRows float64) error {
+	var selectivity float32 = 0.0
+	if totalTableRows != 0.0 {
+		selectivity = float32(float64(actualRows) / totalTableRows)
```
Comment (ghazalfamilyusa) [inline executor/adapter.go:1587]: Can you guard against dividing by zero. In this case, jut ignore the history.

**Review by ghazalfamilyusa — COMMENTED** [review]
Can you unit test the change to cover the restrictions and scope?

File: executor/adapter.go [hunk 2]
```diff
@@ -1504,84 +1505,97 @@ func resetCTEStorageMap(se sessionctx.Context) error {
...
 		}
 	}
-	return false
+	return true
 }
 
-func recordHistoryStats(sql string, tableID int64, planID int, nullHashCode bool, hashCode uint64, estRows float64, actualRows int64) {
+func recordHistoryStats(statsHandle *handle.Handle, startTS uint64, tableID int64, hashCode uint64, actualRows int64, totalTableRows float64) error {
+	var selectivity float32 = 0.0
+	if totalTableRows != 0.0 {
```
Comment (ghazalfamilyusa) [inline executor/adapter.go]: Can you exit if totalTableRows == 0.0? This case is not interesting to capture.

**Review by ghazalfamilyusa — CHANGES_REQUESTED** [review]

**Comment by yibin87** [comment]
> Can you unit test the change to cover the restrictions and scope?

Ok, will add it.

File: executor/adapter.go (hunk 2 above)
Comment (yibin87) [reply executor/adapter.go]: Will change it.

**Review by ghazalfamilyusa — APPROVED** [review]
Will be helpful to add test cases for table scan operator with filter in addition to selection.

**Pull request merged**